cmd/insights/contributors.go

```diff
@@ -0,0 +1,289 @@
+			reposFlag := cmd.Flags().Lookup(common.FlagNameRepos)
+			if !fileFlag.Changed && !reposFlag.Changed {
+				return fmt.Errorf("must pass either %s or %s flag", fileFlag.Name, reposFlag.Name)
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx, apiClient, err := api.DefaultClientSetup(cmd)
+			if err != nil {
+				return err
+			}
+			opts.APIClient = apiClient
+			return run(ctx, opts)
+		},
+	}
+	cmd.Flags().StringVarP(&opts.File, common.FlagNameFile, "f", "", "Path to yaml file containing an array of git repository urls")
+	cmd.Flags().StringSliceVarP(&opts.Repos, common.FlagNameRepos, "r", nil, "Comma-separated array of git repository urls")
+	cmd.Flags().Int32VarP(&opts.Period, common.FlagNamePeriod, "p", 30, "Number of days, used for query filtering")
+	cmd.MarkFlagsMutuallyExclusive(common.FlagNameFile, common.FlagNameRepos)
+	return cmd
+}
+
+func run(ctx context.Context, opts *Options) error {
+	repositories, err := common.HandleRepositoryValues(opts.Repos, opts.File)
+	if err != nil {
+		return err
+	}
+	var (
+		errorChan = make(chan error)
+		doneChan  = make(chan bool)
+		waitGroup = new(sync.WaitGroup)
+	)
+	for repoURL := range repositories {
+		waitGroup.Add(1)
+		go func(repoURL string) {
+			defer waitGroup.Done()
+			repoContributorInsights, err := findAllRepositoryContributors(ctx, opts, repoURL)
+			if err != nil {
+				errorChan <- err
+				return
+			}
+			if repoContributorInsights == nil {
+				return
+			}
+			repoContributorInsights.RenderTable()
+		}(repoURL)
+	}
+	go func() {
+		waitGroup.Wait()
+		close(doneChan)
+	}()
+	select {
+	case <-doneChan:
+		return nil
+	case err = <-errorChan:
+		return err
+	}
+}
+
+type repositoryContributorsInsights struct {
+	RepoID  int32
+	RepoURL string
+	New     []string
+	Recent  []string
+	Churn   []string
+	Repeat  []string
+}
+
+func (rci *repositoryContributorsInsights) RenderTable() {
+	rows := []table.Row{
+		{"New contributors", strconv.Itoa(len(rci.New))},
+		{"Recent contributors", strconv.Itoa(len(rci.Recent))},
+		{"Churn contributors", strconv.Itoa(len(rci.Churn))},
+		{"Repeat contributors", strconv.Itoa(len(rci.Repeat))},
+	}
+	columns := []table.Column{
+		{
+			Title: "Repository URL",
+			Width: 20,
+		},
+		{
+			Title: rci.RepoURL,
+			Width: len(rci.RepoURL),
+		},
+	}
+	styles := table.DefaultStyles()
+	styles.Header.MarginTop(1)
+	styles.Selected = lipgloss.NewStyle()
+	repoTable := table.New(
+		table.WithColumns(columns),
+		table.WithRows(rows),
+		table.WithHeight(len(rows)),
+		table.WithStyles(styles),
+	)
+	fmt.Println(repoTable.View())
+}
+
+type ErrRepoNotFoundOrIndexed struct {
+	RepoURL string
+}
+
+func (e *ErrRepoNotFoundOrIndexed) Error() string {
+	return fmt.Sprintf("repository %s is either non-existent or has not been indexed yet", e.RepoURL)
+}
+
+func NewErrRepoNotFoundOrIndexed(repoURL string) *ErrRepoNotFoundOrIndexed {
+	return &ErrRepoNotFoundOrIndexed{
+		RepoURL: repoURL,
+	}
+}
```

Comment:
Having a structured error here seems unnecessary. Typically I've only seen these to be useful when different error types need to be compared across callers of APIs. Like when using `stat` and `os`:
```go
_, err := os.Stat(path)
if os.IsNotExist(err) { 
    // handle the file at path not existin
}
```
but it doesn't seem we're doing that here. Let's remove this struct type in favor of traditional stacked error handling in Go.